### DISCUSSION TITLE
Milvus-pulsar-recovery is injesting more logs on AKS Log analytics

### DISCUSSION THREAD

**Discussion by manoj555manu** [milvus-io/milvus#30802] [Q&A and General discussion]
Hi,

we have milvus 3.2.1 on AKS 1.26.6.

Seeing an increase of the the below error on "milvus-pulsar-recovery" Pod but application is working fine.

[bookkeeper-io-3-4] ERROR org.apache.bookkeeper.common.allocator.impl.ByteBufAllocatorImpl - Unable to allocate memory
io.netty.util.internal.OutOfDirectMemoryError: failed to allocate 16777216 byte(s) of direct memory (used: 67108864, max: 67108864)

would like to get an insight on this. Thanks in advance.

**Comment by xiaofan-luan** [comment]
Seems that bookie is oom so 
1. https://milvus.io/docs/scale-dependencies.md try to assign more memories to bookie node see if it works
2. check if you are trying to insert in batch, try to reduce the batch size during insert (Recommendation is 1-4MB per batch)

**Reply by manoj555manu** [reply]
Hi @xiaofan-luan,

Thanks for the response.The memory less than15%

**Reply by xiaofan-luan** [reply]
pulsar has MaxDirectMemorySize limitation and seems as yhmo said, you probably need to tune that param

**Comment by yhmo** [comment]
I believe this is a Pulsar issue.
https://github.com/apache/pulsar/issues/9988
Reset the MaxDirectMemorySize value and restart the bookie might work.

**Reply by manoj555manu** [reply]
Hi @yhmo 

we doubled the MaxDirectMemorySize, still seeing the error 

"[bookkeeper-io-3-4] ERROR org.apache.bookkeeper.common.allocator.impl.ByteBufAllocatorImpl - Unable to allocate memory"

 and 
both the liveness and readiness checks failing on  Zookeeper pod
"[Liveness probe failed: ZK server is not ok]"
"[Readiness probe failed: ZK server is not ok]"

**Reply by yhmo** [reply]
Double-check the error message "io.netty.util.internal.OutOfDirectMemoryError: failed to allocate xxxxxxxx byte(s) of direct memory (used: xxxxxxxx, max: xxxxxxx)" to see if the "max" value is changed according to your configuration. Increase the MaxDirectMemorySize until this error message doesn't display.
